index.js

```diff
@@ -441,6 +462,122 @@ class SuperExpressive {
     return next;
   }
 
+  static [mergeSubexpression](el, options, parent, incrementCaptureGroups) {
+    let nextEl = deepCopy(el);
+
+    if (nextEl.type === 'backreference') {
+      nextEl.metadata += parent.state.totalCaptureGroups;
+      incrementCaptureGroups();
```

Comment:
This isn't right - a backreference shouldn't increase the total number of capture groups. Need an additional check below for `capture` which would increase the count.